### PR TITLE
Add optional TwelveLabs Pegasus cloud captioning backend (--backend pegasus)

### PR DESCRIPTION
**Pull request by mohit-twelvelabs**
Hi! I'm Mohit, I work at TwelveLabs (@mohit-twelvelabs).

This adds an **optional** cloud captioning backend powered by [TwelveLabs Pegasus](https://twelvelabs.io), a video-native understanding model, alongside the existing local Qwen-VL / Gemma 4 engines.

### What it adds
- New `pegasus_backend.py` with a `PegasusEngine` that duck-types the subset of `QwenEngine` the CLI uses (`find_files` / `load_model` / `unload_model` / `generate_batch` + the `is_gguf`/`model` sentinels), so it drops into the existing caption loop with no changes to local-model code.
- CLI flags `--backend {local,pegasus}` (default `local`) and `--tl-model` (default `pegasus1.5`).
- It uploads each local video as a TwelveLabs asset, waits for it to be ready, then calls `analyze(...)` and writes the caption with the same trigger-word / `.txt` conventions the local engine uses.

### Why it helps this project
VisionCaptioner currently captions video by sampling ~8 frames and feeding them to a local VLM, which drops most of the motion and temporal ordering in a clip. Pegasus ingests the whole video server-side and reasons over the actual action and sequence of events — often a better fit for captioning **video** training datasets where motion matters. It also needs no GPU/torch, so it's usable on machines that can't run the larger local models.

### Opt-in and non-breaking
- Default stays `--backend local`; nothing changes unless you ask for Pegasus.
- The `twelvelabs` SDK is imported lazily, so users who never select it need no new dependency. It's listed only as a commented optional line in `requirements.txt`.
- Pegasus is video-only; images return a clear per-item error so they're skipped (route images through the local backend). Mixed folders degrade cleanly.

### How it was tested
- `tests/test_pegasus_backend.py`: 11 no-network unit tests mocking the SDK (file discovery, skip/mask handling, missing-key + client construction, `max_tokens` clamp to the model's 512 minimum, trigger-word prepend, image rejection, per-item error capture).
- 1 live connectivity test gated on `TWELVELABS_API_KEY` (skipped without it) asserting a 512-dim Marengo embedding.
- Full suite: **176 passed, 1 skipped**. I also verified the Pegasus `analyze` request wiring against the live API (requests reach the model and validate server-side); a full end-to-end caption on a long video can be slow, so the unit tests cover the wiring deterministically.

Usage:
```bash
export TWELVELABS_API_KEY="tlk_..."
python cli.py --folder /path/to/videos --backend pegasus --prompt "Describe the action." --max-tokens 512
```

You can grab a free API key at https://twelvelabs.io — there's a generous free tier.

### CODE REVIEW

**Comment by Brekel** [comment]
Hi Mohit — thanks for the work here, the integration looks clean and the test coverage is appreciated.

I'm sorry but I'm going to decline this one. VisionCaptioner is deliberately a local-first, self-contained tool, and I'd rather not ship a commercial provider's service as a captioning backend, even an optional one. It ties the project to a vendor I'm not affiliated with, leaves me maintaining an integration against an API that can change, and the flags and docs inevitably read as an endorsement.

That's not a comment on Pegasus or on the code quality. If there's demand for it, a fork or a separate plugin repo is a good home.

Thanks again for taking the time.
